### PR TITLE
UCS/SYS: THP sysfs parsing

### PR DESCRIPTION
**Pull request by eladpe**
Fix bug in THP parsing

### CODE REVIEW

File: src/ucs/sys/sys.c [hunk 1]
```diff
@@ -411,7 +411,7 @@ int ucs_is_thp_enabled()
         return 0;
     }
 
-    return (strcmp(buf, "always madvise [never]") != 0);
+    return (strcmp(buf, "always madvise [never]\n") != 0);
```
Comment (yosefe) [inline src/ucs/sys/sys.c]: this is not portable fix

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3938/ for details (Mellanox internal link).

**Comment by eladpe** [comment]
@yosefe ready for review

File: src/ucs/sys/sys.c [hunk 2]
```diff
@@ -404,14 +404,15 @@ int ucs_is_thp_enabled()
...
+    char *never = "always madvise [never]";
 
     rc = ucs_read_file(buf, sizeof(buf), 1, UCS_SYS_THP_ENABLED_FILE);
     if (rc < 0) {
         ucs_debug("failed to read %s:%m", UCS_SYS_THP_ENABLED_FILE);
         return 0;
     }
 
-    return (strcmp(buf, "always madvise [never]") != 0);
+    return (strncmp(buf, never, strlen(never)) != 0);
```
Comment (yosefe) [inline src/ucs/sys/sys.c]: better do strstr(buf, "[never]")

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3939/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3941/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3943/ for details (Mellanox internal link).

**Comment by eladpe** [comment]
@yosefe ready for review (2)

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Pull request merged**